packages/feature-toggle/src/withFeature/withFeature.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react'
+import useFeature from '../useFeatures/useFeature'
+import { nameOf } from '../react-utils'
+
+// Need this to land in TypeScript first for better type inference
+// Link: https://github.com/Microsoft/TypeScript/issues/26242
+const withFeature = <T extends object, TComponentProps = {}>(
+  Component: React.ComponentType<TComponentProps>,
+  name: keyof T
+) => {
+  type OwnProps = Omit<TComponentProps, keyof T>
+
+  const Wrapped: React.FC<OwnProps> = React.memo(props => {
+    const feature = useFeature<T>()[name]
```

Comment:
Then we would use it as `useFeature<T>(name)`

Reply:
☝️